clientconn.go

```diff
@@ -146,12 +146,14 @@ func DialContext(ctx context.Context, target string, opts ...DialOption) (conn *
 	cc.safeConfigSelector.UpdateConfigSelector(&defaultConfigSelector{nil})
 	cc.ctx, cc.cancel = context.WithCancel(context.Background())
 
-	for _, opt := range extraDialOptions {
+	for _, opt := range opts {
```

Comment:
This also rearranges the order in which the options are applied.

I believe the old way was intentional (global things come first so local ones can override).  Is there a reason you changed this or was it unintended?

Reply:
I changed this because it would be inefficient to loop through all of the options (o(n)) to determine whether to disable global options, and then if disable { for each global options}     then for each options. I actually had a thought about just using the overwrite logic (way it was in master) to plumb in the filtered out stats handler/set no logger, but the problem with that is I wouldn't know which logger to overwrite (there's two, and an extra would just append). I don't think there's a use case where ordering matters here (from the users perspective, all callsites are o11y), even global options are marked internal.